numpy/lib/tests/test_function_base.py

```diff
@@ -1499,6 +1500,43 @@ def test_size_zero_output(self):
         with assert_raises_regex(ValueError, 'new output dimensions'):
             f(x)
 
+    @pytest.mark.skipif(not HAS_REFCOUNT, reason="Python lacks refcounts")
+    def test_frompyfunc_leaks(self):
+        # exposed in gh-11867 as np.vectorized, but the problem stems from
+        # frompyfunc.
+        # class.attribute = np.frompyfunc(<method>) creates a
+        # reference cycle that requires a gc collection cycle to break
+        # (on CPython 3)
+        import gc
+        gc.disable()
+        iters = 20
+        try:
+            class A(object):
+                def method(self, *args):
+                    return 0
+
+                def __init__(self):
+                    self.f = np.frompyfunc(self.method, 1, 1)
+
+            refcount = sys.getrefcount(A.method)
+            for i in range(iters):
+                a = A()
+                out = a.f(0)
+                a.f_unbound = np.frompyfunc(A.method, 2, 1)
+                out = a.f_unbound(0, 0)
```

Comment:
Can you separate the bound and unbound tests?

Reply:
done